### ISSUE TITLE
startTime.preciseDiff is not a function

### ISSUE THREAD

**Issue by OmgImAlexis** [decaffeinate/bulk-decaffeinate#140]
Issue started in https://github.com/decaffeinate/bulk-decaffeinate/commit/1a26aa73166ae9997be31f4d6a95ea203c303acd

```
➜  dredd git:(master) bulk-decaffeinate check
Doing a dry run of decaffeinate on 97 files... (8 workers)
97/97
TypeError: startTime.preciseDiff is not a function
    at runWithProgressBar$ (/Users/xo/.config/yarn/global/node_modules/bulk-decaffeinate/dist/bulk-decaffeinate.js:1543:51)
    at tryCatch (/Users/xo/.config/yarn/global/node_modules/regenerator-runtime/runtime.js:65:40)
    at Generator.invoke [as _invoke] (/Users/xo/.config/yarn/global/node_modules/regenerator-runtime/runtime.js:303:22)
    at Generator.prototype.(anonymous function) [as next] (/Users/xo/.config/yarn/global/node_modules/regenerator-runtime/runtime.js:117:21)
    at tryCatch (/Users/xo/.config/yarn/global/node_modules/regenerator-runtime/runtime.js:65:40)
    at invoke (/Users/xo/.config/yarn/global/node_modules/regenerator-runtime/runtime.js:155:20)
    at /Users/xo/.config/yarn/global/node_modules/regenerator-runtime/runtime.js:165:13
    at <anonymous>
```

**Comment by alangpierce** [comment]
Thanks for reporting. Unfortunately, it seems to be working for me. From the stack trace, looks like you have bulk-decaffeinate globally installed via yarn on macOS, which is the configuration I used as well.

From the commit you linked, `import 'moment-precise-range-plugin';` is supposed to have a side-effect of attaching `preciseDiff` to the moment object prototype, but it looks like somehow that's not happening here. It looks like that plugin has a direct dependency on the `moment` package, meaning that depending on how yarn is feeling it's possible that it would get its own copy of moment rather than modifying bulk-decaffeinate's copy. Probably the plugin should have used a peer dependency instead.

Maybe try reinstalling bulk-decaffeinate? (`yarn global remove bulk-decaffeinate` and then `yarn global add bulk-decaffeinate`). Or maybe try installing it within your project (not globally) or using npm or some other configuration like that?

Another thing you can do is check if `moment-precise-range-plugin` is properly installed and has no explicit dependencies. For me, `/Users/alanpierce/.config/yarn/global/node_modules/moment-precise-range-plugin` does not have any `node_modules` subdirectory, meaning it'll use the shared `moment` instance.

Other questions that might help if that doesn't work:
* What node and yarn version are you using?
* Did you have bulk-decaffeinate installed before the October 28 diff that you linked?

**Comment by OmgImAlexis** [comment]
I'm using `node v8.9.1` and `yarn 1.3.2`.
I installed `bulk-decaffeinate` as well as `decaffeinate` about 10 hours ago. I haven't used either of them before.

```
➜  dredd git:(master) yarn global remove bulk-decaffeinate
yarn global v1.3.2
[1/2] Removing module bulk-decaffeinate...
[2/2] Regenerating lockfile and installing missing dependencies...
warning " > generator-chrome-extension-kickstart@4.4.2" has incorrect peer dependency "yo@1.x.x".
success Uninstalled packages.
✨  Done in 276.18s.
➜  dredd git:(master) yarn global add bulk-decaffeinate
yarn global v1.3.2
[1/4] 🔍  Resolving packages...
warning bulk-decaffeinate > fs-promise@1.0.0: Use mz or fs-extra^3.0 with Promise Support
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
warning " > generator-chrome-extension-kickstart@4.4.2" has incorrect peer dependency "yo@1.x.x".
[4/4] 📃  Building fresh packages...
success Installed "bulk-decaffeinate@3.3.0" with binaries:
      - bulk-decaffeinate
✨  Done in 33.12s.
```

```
➜  dredd git:(master) bulk-decaffeinate check
Doing a dry run of decaffeinate on 97 files... (8 workers)
97/97
TypeError: startTime.preciseDiff is not a function
    at runWithProgressBar$ (/Users/xo/.config/yarn/global/node_modules/bulk-decaffeinate/dist/bulk-decaffeinate.js:1543:51)
    at tryCatch (/Users/xo/.config/yarn/global/node_modules/regenerator-runtime/runtime.js:65:40)
    at Generator.invoke [as _invoke] (/Users/xo/.config/yarn/global/node_modules/regenerator-runtime/runtime.js:303:22)
    at Generator.prototype.(anonymous function) [as next] (/Users/xo/.config/yarn/global/node_modules/regenerator-runtime/runtime.js:117:21)
    at tryCatch (/Users/xo/.config/yarn/global/node_modules/regenerator-runtime/runtime.js:65:40)
    at invoke (/Users/xo/.config/yarn/global/node_modules/regenerator-runtime/runtime.js:155:20)
    at /Users/xo/.config/yarn/global/node_modules/regenerator-runtime/runtime.js:165:13
    at <anonymous>
```

```
➜  dredd git:(master) ls ~/.config/yarn/global/node_modules/moment-precise-range-plugin
LICENSE.md              README.md               moment-precise-range.js node_modules            package.json
➜  dredd git:(master) ls ~/.config/yarn/global/node_modules/moment-precise-range-plugin/node_modules 
moment
```

**Comment by OmgImAlexis** [comment]
Removing the moment dir from `node_modules` doesn't seem to make a difference.

```
➜  dredd git:(master) mv ~/.config/yarn/global/node_modules/moment-precise-range-plugin/node_modules/moment ~/.config/yarn/global/node_modules/moment-precise-range-plugin/node_modules/moment_old
➜  dredd git:(master) bulk-decaffeinate check                                                                                                                                                     
Doing a dry run of decaffeinate on 97 files... (8 workers)
97/97
TypeError: startTime.preciseDiff is not a function
    at runWithProgressBar$ (/Users/xo/.config/yarn/global/node_modules/bulk-decaffeinate/dist/bulk-decaffeinate.js:1543:51)
    at tryCatch (/Users/xo/.config/yarn/global/node_modules/regenerator-runtime/runtime.js:65:40)
    at Generator.invoke [as _invoke] (/Users/xo/.config/yarn/global/node_modules/regenerator-runtime/runtime.js:303:22)
    at Generator.prototype.(anonymous function) [as next] (/Users/xo/.config/yarn/global/node_modules/regenerator-runtime/runtime.js:117:21)
    at tryCatch (/Users/xo/.config/yarn/global/node_modules/regenerator-runtime/runtime.js:65:40)
    at invoke (/Users/xo/.config/yarn/global/node_modules/regenerator-runtime/runtime.js:155:20)
    at /Users/xo/.config/yarn/global/node_modules/regenerator-runtime/runtime.js:165:13
    at <anonymous>
```

**Comment by alangpierce** [comment]
Thanks for the info. The existence of the separate moment installation definitely suggests the plugin is modifying the wrong moment instance. Unfortunately, I still can't repro, that's annoying. 😕

My current guess: some existing globally-installed package had already installed some version of moment, maybe moment 1.x. Yarn needed to install bulk-decaffeinate and moment-precise-range-plugin and both of those have version ranges specifying moment 2.x, so yarn decided to do a separate moment installation for each, so that's 3 moment installations total. When you renamed the node_modules package, the plugin switched from modifying its own moment and instead modified the top-level moment, but it still wasn't modifying bulk-decaffeinate's moment.

I sent a PR to change the dependency to a peer dependency https://github.com/codebox/moment-precise-range/pull/32 , but I'm still a little unsure if that would fix this case.

Given the fragility of moment-precise-range-plugin, I think I'll just release a change to drop that dependency and copy the relevant source code to my project, at least for now. I think I should be able to get that released in the next few hours.

**Comment by alangpierce** [comment]
Ok, I just released bulk-decaffeinate 3.3.1, which hopefully should fix the issue.
